gulpfile.js

```diff
@@ -1,3 +1,18 @@
+// check the minimal supported version of node
+const pkg = require('./package.json');
+const gutil = require('gulp-util');
+
+const nodeMajorVersion = process.version.substr(1, 1);
+const minMajorVersion = pkg.engines.node.substr(-1);
+
+if (nodeMajorVersion < minMajorVersion) {
+    gutil.log(
+        gutil.colors.red(`Node version ${pkg.engines.node} is required.`),
+        gutil.colors.cyan(`You are using ${process.version}`)
+    );
+    process.exit(1);
+}
+
```

Comment:
What about moving the whole check code to a function exported eg. from `gulp/check-compatibility.js` and then just calling `require('./check-compatibility')()`?